ui/page/components/account_selector.go

```diff
@@ -200,34 +203,44 @@ func (as *AccountSelector) Layout(gtx layout.Context) layout.Dimensions {
 }
 
 func (as *AccountSelector) ListenForTxNotifications(ctx context.Context) {
+	if as.TxAndBlockNotificationListener == nil {
+		as.TxAndBlockNotificationListener = listeners.NewTxAndBlockNotificationListener()
+	}
+
+	err := as.WL.MultiWallet.AddTxAndBlockNotificationListener(as.TxAndBlockNotificationListener, true, AccoutSelectorID)
+	if err != nil {
+		log.Errorf("Error Adding Tx notifications listener: %+v", err)
+		return
+	}
+
 	go func() {
 		for {
-			var notification interface{}
-
 			select {
-			case notification = <-as.Receiver.NotificationsUpdate:
-			case <-ctx.Done():
-				return
-			}
-
-			switch n := notification.(type) {
-			case wallet.NewTransaction:
-				// refresh accounts list when new transaction is received
-				as.UpdateSelectedAccountBalance()
-				if as.selectorModal != nil {
-					as.selectorModal.setupWalletAccounts()
-				}
-				as.RefreshWindow()
-			case wallet.SyncStatusUpdate:
-				// refresh wallet account and balance on every new block
-				// only if sync is completed.
-				if n.Stage == wallet.BlockAttached && as.WL.MultiWallet.IsSynced() {
+			case n := <-as.TxAndBlockNotifChan:
+				switch n.Type {
+				case listeners.BlockAttached:
+					// refresh wallet account and balance on every new block
+					// only if sync is completed.
+					if as.WL.MultiWallet.IsSynced() {
+						as.UpdateSelectedAccountBalance()
+						if as.selectorModal != nil {
+							as.selectorModal.setupWalletAccounts()
+						}
+						as.RefreshWindow()
+					}
+				case listeners.NewTransaction:
+					// refresh accounts list when new transaction is received
 					as.UpdateSelectedAccountBalance()
 					if as.selectorModal != nil {
 						as.selectorModal.setupWalletAccounts()
 					}
 					as.RefreshWindow()
 				}
+			case <-ctx.Done():
+				as.WL.MultiWallet.RemoveTxAndBlockNotificationListener(AccoutSelectorID)
+				// Close notification channel.
```

Comment:
Not sure this comment is necessary. It's quite obvious what's happening in the next line.